src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -304,7 +304,7 @@ public override void AttachToDependencyGraph(DependencyAnalyzerBase<NodeFactory>
                 CorCompileImportType.CORCOMPILE_IMPORT_TYPE_STUB_DISPATCH,
                 CorCompileImportFlags.CORCOMPILE_IMPORT_FLAGS_PCODE,
                 (byte)Target.PointerSize,
-                emitPrecode: false,
+                emitPrecode: true,
```

Comment:
The precode fixup fixup should be emitted only when the method is used as delegate target. If it is a regular call, it should be delay-load call.

Reply:
Attempting to fix it by separating the method imports into separate tables. It is still a work in progress. I have got something that passes the `tests\runtest.cmd /mode readytorun` and the regressed JIT test case.